### PR TITLE
Add deprecation precheck

### PR DESCRIPTION
**Pull request by tiagonapoli**
#### What is the purpose of this pull request?
Installing toolbelt will just warn about the deprecation and will allow installation.

This PR will deactivate the installed version if it's deprecated

#### How should this be manually tested?
```
git clone https://github.com/vtex/toolbelt.git && \
cd toolbelt && \
git checkout fix/check-npm-deprecation && \
yarn && yarn global add file:$PWD
```
Change the version on `package.json` to `2.86.0`:
```
code  $(yarn global dir)/node_modules/vtex/package.json
```
Try to run `vtex whoami` - it will warn about deprecation and close the process.

#### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -22,51 +21,19 @@ import { Token } from './Token.js'
...
-const loginCmd = tree.login
-let loginPending = false
-
-if (process.env.NODE_ENV === 'development') {
-  try {
-    require('longjohn') // tslint:disable-line
-  } catch (e) {
-    log.debug("Couldn't require longjohn. If you want long stack traces, run: npm install -g longjohn")
-  }
-}
```
Comment (tiagonapoli) [inline src/cli.ts]: This was removed

File: src/cli.ts [hunk 2]
```diff
@@ -22,51 +21,19 @@ import { Token } from './Token.js'
...
 import { isVerbose, VERBOSE } from './utils'
 
-const nodeVersion = process.version.replace('v', '')
-if (!semver.satisfies(nodeVersion, pkg.engines.node)) {
-  const minMajor = pkg.engines.node.replace('>=', '')
-  console.error(
-    chalk.bold(`Incompatible with node < v${minMajor}. Please upgrade node to major ${minMajor} or higher.`)
-  )
-  process.exit(1)
-}
```
Comment (tiagonapoli) [inline src/cli.ts]: This was moved to CLIPrechecker

File: src/cli.ts [hunk 3]
```diff
@@ -22,51 +21,19 @@ import { Token } from './Token.js'
...
-  if (envCookies()) {
-    config.headers.Cookie = `${envCookies()}; ${config.headers.Cookie || ''}`
-  }
-  return config
-})
-
-global.Promise = Bluebird
-Bluebird.config({
-  cancellation: true,
-})
```
Comment (tiagonapoli) [inline src/cli.ts]: This was moved inside this same file

File: src/cli.ts [hunk 4]
```diff
@@ -22,51 +21,19 @@ import { Token } from './Token.js'
...
-
 const run = command => Bluebird.resolve(unboundRun.call(tree, command, path.join(__dirname, 'modules')))
 
-const loginCmd = tree.login
-let loginPending = false
```
Comment (tiagonapoli) [inline src/cli.ts]: This was moved inside this same file

File: src/cli.ts [hunk 5]
```diff
@@ -22,51 +21,19 @@ import { Token } from './Token.js'
...
-if (process.env.NODE_ENV === 'development') {
-  try {
-    require('longjohn') // tslint:disable-line
-  } catch (e) {
-    log.debug("Couldn't require longjohn. If you want long stack traces, run: npm install -g longjohn")
-  }
-}
-
-// Show update notification if newer version is available
-notify()
```
Comment (tiagonapoli) [inline src/cli.ts]: This was moved inside this same file

File: src/CLIPrechecker.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      console.error(
+        chalk.bold(
+          `This version ${pkg.version} was deprecated. Please update to the latest version: ${chalk.green(
+            'yarn global add vtex'
+          )}.`
+        )
+      )
+
+      process.exit(1)
+    }
```
Comment (dhasuda) [inline src/CLIPrechecker.ts]: _nit_ to avoid indenting: you can make
```
if (deprecated == null) return
```

File: src/CLIPrechecker.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  private static ensureCompatibleNode() {
+    const nodeVersion = process.version
+    if (!semver.satisfies(nodeVersion, pkg.engines.node)) {
+      const minMajor = pkg.engines.node.replace('>=', '')
+      console.error(
+        chalk.bold(`Incompatible with node < v${minMajor}. Please upgrade node to major ${minMajor} or higher.`)
+      )
+
+      process.exit(1)
+    }
```
Comment (dhasuda) [inline src/CLIPrechecker.ts]: _Refactor suggestion_:
```
const pkgNodeVersion = pkg.engines.node
if (semver.satisfies(nodeVersion, pkgNodeVersion) { return }
const minMajor = pkgNodeVersio.replace('>=', '')
const errorMsg = chalk.bold(`Incompatible with node < v${minMajor}. Please upgrade node to major ${minMajor} or higher.`
console.error(errorMsg)

process.exit(1)
```

**Pull request merged**